test/widgets/message_list_test.dart

```diff
@@ -1182,6 +1182,120 @@ void main() {
       return finder.evaluate().isNotEmpty;
     }
 
+    (Widget, Widget) checkConfirmDialog(WidgetTester tester, int unreadCount) {
+      final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+      return checkSuggestedActionDialog(tester,
+        expectedTitle: zulipLocalizations.markAllAsReadConfirmationDialogTitle,
+        expectedMessage: zulipLocalizations.markAllAsReadConfirmationDialogMessage(unreadCount),
+        expectDestructiveActionButton: false,
+        expectedActionButtonText: zulipLocalizations.markAllAsReadConfirmationDialogConfirmButton,
+      );
+    }
+
+    group('confirmation dialog', () {
+      Future<void> showsConfirmDialog({required WidgetTester tester,
+        required Narrow narrow, required List<Message> messages,
+        required UnreadMessagesSnapshot unreadMsgs,
+      }) async {
+        await setupMessageListPage(tester, narrow: narrow,
+          messages: messages, unreadMsgs: unreadMsgs);
+        check(isMarkAsReadButtonVisible(tester)).isTrue();
+
+        await tester.tap(find.byType(MarkAsReadWidget));
+        await tester.pump();
+
+        final unreadCount = store.unreads.countInNarrow(narrow);
+        checkConfirmDialog(tester, unreadCount);
+      }
+
+      Future<void> doesNotShowConfirmDialog({required WidgetTester tester,
+        required Narrow narrow, required List<Message> messages,
+        required UnreadMessagesSnapshot unreadMsgs,
+        required UpdateMessageFlagsForNarrowResult updateResult,
+      }) async {
+        await setupMessageListPage(tester,
+          narrow: narrow, messages: messages, unreadMsgs: unreadMsgs);
+
+        check(isMarkAsReadButtonVisible(tester)).isTrue();
+
+        connection.prepare(json: updateResult.toJson());
+        await tester.tap(find.byType(MarkAsReadWidget));
+        await tester.pump();
+
+        checkNoDialog(tester);
+        await tester.pump(Duration.zero);
+      }
+
+      testWidgets('CombinedFeedNarrow: show dialog', (tester) async {
+        final message = eg.streamMessage(flags: []);
+        final unreadMsgs = eg.unreadMsgs(channels:[
```

Comment:
```suggestion
        final unreadMsgs = eg.unreadMsgs(channels: [
```

(here and in some other tests)